R/timeseries_plot.R

```diff
@@ -0,0 +1,98 @@
+#' Title
+#'
+#' @param data 
+#' @param time 
+#' @param elements 
+#' @param station 
+#' @param facets 
+#' @param add_points 
+#' @param add_line_of_best_fit 
+#' @param se 
+#' @param add_path 
+#' @param add_step 
+#' @param na.rm 
+#' @param show.legend 
+#'
+#' @return
+#' @export
+#'
+#' @examples
+timeseries_plot <- function(data, time, elements, station = NULL, facets = c("none", "elements", "stations", "both"),
+                            add_points = FALSE, add_line_of_best_fit = FALSE,
+                            se = TRUE, add_path = FALSE, add_step = FALSE,
+                            na.rm = FALSE, show.legend = NA){
+  
+  checkmate::assert_data_frame(data)
+  # time can be a date, factor, or character.
+  # call date_time for consistency with summary plot?
+  checkmate::assert_character(elements)
+  checkmate::assert_character(station, null.ok = TRUE)
+  facets <- match.arg(facets)
+  checkmate::assert_logical(add_points)
+  checkmate::assert_logical(add_line_of_best_fit)
+  checkmate::assert_logical(se)
+  checkmate::assert_logical(add_path)
+  checkmate::assert_logical(add_step)
+  checkmate::assert_logical(show.legend)
+  
+  data_longer <- data %>% tidyr::pivot_longer(cols = c({{ elements }}), names_to = "elements")
```

Comment:
Shouldn't this be `all_of(elements)` if `elements` is a character?

Reply:
Great. I was trying to figure this out using the string function technique (instead of the embraces)